src/harness/testcases/grant_testcase.py

```diff
@@ -285,6 +288,72 @@ def test_WINNF_FT_S_GRA_14(self):
     self.assertEqual(response['response']['responseCode'], 103)
 
   @winnforum_testcase
+  def test_WINNF_FT_S_GRA_15(self):
+    """CBSD requests a frequency range which is a mix of PAL and GAA channel.
+    
+    The Response Code should be 103(INVALID_PARAM)
+    """
+
+    # Load the Data
+    pal_low_frequency = 3550000000.0
+    pal_high_frequency = 3560000000.0
+    device_a = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    pal_record = json.load(
+      open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+    ppa_record = json.load(
+      open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+    ppa_record, pal_record = makePpaAndPalRecordsConsistent(ppa_record,
+                                                            [pal_record],
+                                                            pal_low_frequency,
+                                                            pal_high_frequency,
+                                                            device_a['userId'])
+
+    # Move the Device to a random location in PPA
+    device_a['installationParam']['latitude'], \
+    device_a['installationParam']['longitude'] = getRandomLatLongInPolygon(ppa_record)
+
+    # Register the device
+    self._sas_admin.InjectFccId({'fccId': device_a['fccId']})
+    request = {'registrationRequest': [device_a]}
+    response = self._sas.Registration(request)['registrationResponse'][0]
+    # Check registration response
+    self.assertEqual(response['response']['responseCode'], 0)
+    cbsd_id = response['cbsdId']
+    del request, response
+
+    # Inject PAL and PPA Database Record
+    ppa_record['ppaInfo']['cbsdReferenceId'] = [cbsd_id]
+    self._sas_admin.InjectPalDatabaseRecord(pal_record[0])
+    self._sas_admin.InjectZoneData({'record': ppa_record})
+
+    # Trigger daily activities and wait for it to get it complete
+    self._sas_admin.TriggerDailyActivitiesImmediately()
```

Comment:
My thought was something like "TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()" (kind of your #1 but without the Scheduled option).